### PR TITLE
Add sync API for transactional client

### PR DESCRIPTION
**Pull request by odecode**
For issue #289  a synchronous API for transactional client.

I have programmed a fair bit but new to OSS and Rust. I consulted AI heavily for this to try and get an understanding of what this part of the project is about and how it works, but I tried using it a teacher rather than just "vibe coding".

## Summary by CodeRabbit

* **New Features**
  * Exposed blocking synchronous transaction and snapshot APIs that provide full, blocking access to transactional and read-only operations alongside existing async interfaces.
* **Bug Fixes**
  * Clear user-facing error when attempting to use the synchronous client from within an active async runtime.
* **Tests**
  * Extensive integration tests covering sync client usage, transactions, snapshots, scans, edge cases, and context validation.

### CODE REVIEW

**Comment by odecode** [comment]
Thank you for your patience so far! I have made the review's suggested change.

File: tests/sync_transaction_tests.rs [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+use tikv_client::Value;
+
+/// Helper to initialize and return a sync client
+fn sync_client() -> Result<SyncTransactionClient> {
+    SyncTransactionClient::new_with_config(pd_addrs(), Config::default().with_default_keyspace())
+}
+
+#[test]
+#[serial]
+fn sync_txn_get_timestamp() -> Result<()> {
```
Comment (pingyu) [inline tests/sync_transaction_tests.rs]: CI will not cover test cases with prefix `sync_txn_`. Please see the `Makefile`. Suggest to name as `txn_sync_`.

File: tests/common/mod.rs [hunk 2]
```diff
@@ -74,6 +74,14 @@ pub async fn init() -> Result<()> {
     Ok(())
 }
 
+pub fn init_sync() -> Result<()> {
```
Comment (pingyu) [inline tests/common/mod.rs]: I think we can simply implement as: `tokio::runtime::Builder::new_current_thread().enable_all().build()?.block_on(init())`.

As we can guarantee that this function must be the first of the test cases and never be called inside a tokio runtime.

It would be even better if we add a comment about the usage limitation.

**Comment by odecode** [comment]
Is it possible for me to request review from Copilot myself? Then I could address all its concerns and not bother you on every iteration.

**Comment by odecode** [comment]
added extra test coverage for the sync snapshot public api surface, and error handling to prevent panic when sync client is called from within an async runtime.

File: src/transaction/sync_transaction.rs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    /// Rollback the transaction.
+    pub fn rollback(&mut self) -> Result<()> {
+        self.runtime.block_on(self.inner.rollback())
+    }
+
+    /// Send a heart beat message to keep the transaction alive.
+    pub fn send_heart_beat(&mut self) -> Result<u64> {
+        self.runtime.block_on(self.inner.send_heart_beat())
+    }
```
[the remark this answers is not in the document]
Comment (odecode) [reply src/transaction/sync_transaction.rs:126]: this is fixed

File: src/common/errors.rs [hunk 4]
```diff
@@ -128,6 +128,12 @@ pub enum Error {
     KeyspaceNotFound(String),
     #[error("Transaction not found error: {:?}", _0)]
     TxnNotFound(kvrpcpb::TxnNotFound),
+    /// Attempted to create or use the sync client (including calling its methods) from within a Tokio async runtime context
```
Comment (pingyu) [inline src/common/errors.rs:131]: `Error::NestedRuntimeError`’s display string already wraps an inner `{0}` with guidance (`src/common/errors.rs:132`), but `check_nested_runtime()` also stores a long, guidance-heavy string (`src/transaction/sync_client.rs:30`). This likely produces duplicated advice and punctuation (e.g., “... any Tokio runtime.. Use TransactionClient ...”), and makes future message changes harder.

Comment (odecode) [reply src/common/errors.rs:131]: I changed it so that the guidance is only in Error::NestedRuntimeError, but left the {0} parameter there so that potential callers may add additional information if needed.

**Comment by pingyu** [comment]
cc @iosmanthus

**Comment by pingyu** [comment]
The build failure will be fixed by #532 .

**Comment by pingyu** [comment]
@odecode Thanks for your contribution !
